xdsl/irdl/constraints.py

```diff
@@ -72,24 +82,23 @@ def verify(
         """
         ...
 
-    def get_resolved_variables(self) -> set[str]:
+    def get_resolvers(self) -> dict[str, Resolver[AttributeCovT]]:
         """
-        Get the set of type variables that are always resolved when verifying
-        the constraint.
+        Get a dictionary of variables that can be solved from this attribute.
         """
-        return set()
+        return dict()
 
-    def can_infer(self, constraint_names: set[str]) -> bool:
+    def can_infer(self, variables: Set[str]) -> bool:
```

Comment:
How do you feel about this?
```suggestion
    def can_infer(self, `var_constraint_names`: Set[str]) -> bool:
```
I probably should have commented this on the previous PR, my bad

Reply:
I assume you didn't mean to include the backticks? It seems reasonable to me.

Reply:
I didn't indeed

Reply:
There was a small chance it was some obscure python syntax I'd not seen before

Reply:
it's obscure Swift and Mojo syntax!